### ISSUE TITLE
Select between "append" and "replace" completion behavior

### ISSUE THREAD

**Issue by matklad** [microsoft/language-server-protocol#793]
If the cursor is in the middle of an identifier, there are two reasonable ways to do completion:

```
function(x) foo {}

function main() {
    fo|quux()
}
```

One is it "append" text to the cursor:

```
function(x) foo {}

function main() {
    foo(|)quux()
}
```

Another one is to replace existing identifier:

```
function(x) foo {}

function main() {
    foo(|)
}
```

In IntelliJ, the behavior depends on the key used for completion (<kbd>Enter</kbd> appends, <kbd>Tab</kbd> replaces). 

I think currently it's impossible to express this feature in the LSP, because `CompletionItem` comes with a fixed text-edit, so the server needs to pick *some* option.

**Comment by Gama11** [comment]
>In IntelliJ, the behavior depends on the key used for completion (<Enter> appends, <Tab> replaces).

That seems related to my suggestion in https://github.com/microsoft/vscode/issues/70824.

**Comment by dbaeumer** [comment]
So the completion item would basically need to provide two different edits.

**Comment by matklad** [comment]
Either that, or resolveCompletionItem could tell the server which variant is required.

**Comment by dbaeumer** [comment]
We started to work on this in VS Code.

**Comment by dbaeumer** [comment]
This got already implemented and documented in the 3.16 spec.
